frank-flow/src/frontend/src/app/shared/services/flow-structure.service.ts

```diff
@@ -207,6 +207,69 @@ export class FlowStructureService {
     );
   }
 
+  setFirstPipeById(firstPipeId: string): void {
+    const firstPipeName = this.flowStructure.pipes?.find(
+      (pipe) => pipe.uid === firstPipeId
+    )?.name;
+
+    if (firstPipeName) {
+      this.setFirstPipe(firstPipeName);
+    }
+  }
+
+  removeFirstPipe(): void {
+    const pipelineAttributes = this.flowStructure.pipeline.attributes;
+    if (pipelineAttributes['firstPipe']?.value) {
+      this.deleteAttribute('firstPipe', pipelineAttributes, true);
+    } else {
+      this.currentFile.flowNeedsUpdate = true;
+      this.currentFileService.updateCurrentFile(this.currentFile);
+    }
+  }
+
+  setFirstPipe(firstPipe: string): void {
+    const pipelineAttributes = this.flowStructure.pipeline.attributes;
+    const attributeName = 'firstPipe';
+
+    if (this.attributeListIsEmpty(pipelineAttributes)) {
+      this.createFirstAttribute(
+        attributeName,
+        firstPipe,
+        this.flowStructure.pipeline,
+        true
+      );
+    } else {
+      const editOperations: monaco.editor.IIdentifiedSingleEditOperation[] = [];
+      const editAttributeOperation = this.editAttribute(
+        attributeName,
+        firstPipe,
+        pipelineAttributes,
+        true
+      );
+      if (editAttributeOperation) {
+        editOperations.push(editAttributeOperation);
+      }
+      this.monacoEditorComponent?.applyEdits(editOperations, true);
+    }
+  }
+
+  createFirstAttribute(
+    attributeName: string,
```

Comment:
attribute name and value are related to eachother, to prevent using 4 parameters you can make a config object for this or personally I think a tuple would be best for this kind of problem.

I know JS doesn't have a tuple but JS does have something simmilar:

https://stackoverflow.com/questions/4512405/javascript-variable-assignments-from-tuples

Reply:
Yes a destructed array

Reply:
I'll do you one better and use a type for it :D